### PR TITLE
bazel: pin legacy images and make them usable with push_all

### PR DESCRIPTION
**Pull request by burmudar**
With this PR the legacy images will not be built in a seperate step anymore and outside of bazel. Instead, they will be built with bazel and with all the other images.

### How to still build legacy images
If you still want to build legacy images you can push your branch to `docker-images-candidates-notest/<yourname here>` or alternatively run `sg ci build docker-images-candidates-notest`

## Test plan
ran push_all locally (with the exit 1) and the images were picked up

### CODE REVIEW

File: dev/oci_deps.bzl [hunk 1]
```diff
@@ -207,3 +207,36 @@ def oci_deps():
         digest = "sha256:dcc08b0228a5abd8a542efc7ee86ec58299591a737cafe19e92af6c1e1301c26",
         image = "index.docker.io/sourcegraph/wolfi-qdrant-base",
     )
+    _legacy_image_deps()
+
+def _legacy_image_deps():
+    oci_pull(
+        name = "legacy_alpine-3.14_base",
+        digest = "sha256:581afabd476b4918b14295ae6dd184f4a3783c64bab8bde9ad7b11ea984498a8",
```
Comment (burmudar) [inline dev/oci_deps.bzl:215]: all image digests are from `252535_2023-11-28_5.2-82b5f4f5d73f`

File: dev/ci/push_all.sh [hunk 2]
```diff
@@ -136,6 +136,7 @@ echo "-- jobfile"
 cat "$job_file"
 echo "--- done"
 
+exit 1
```
Comment (burmudar) [inline dev/ci/push_all.sh:139]: for safety 🧯

File: dev/oci_deps.bzl (hunk 1 above)
Comment (jhchabran) [reply dev/oci_deps.bzl:215]: Would be cool to have a script that deals with this, i.e. refresh those, but it's not like we rebuild them very often, so we're probably fine. 

I wonder though what kind of guardrails we could put in place to spot drift happening here.

**Review by jhchabran — COMMENTED** [review]
NICE

One question remains though, how do we rebuild those images outside of Bazel if we have to?

**Comment by burmudar** [comment]
> NICE
> 
> One question remains though, how do we rebuild those images outside of Bazel if we have to?

We could probably keep the current method that does around. Just have to rename it as it is currently `bazelBuildCandidateDockerImages` and maybe under different runType like `legacyDockerImages`

**Comment by burmudar** [comment]
> > NICE
> > One question remains though, how do we rebuild those images outside of Bazel if we have to?
> 
> We could probably keep the current method that does around. Just have to rename it as it is currently `bazelBuildCandidateDockerImages` and maybe under different runType like `legacyDockerImages`

@jhchabran So this is exactly what the CandidateNoTest runtype does. So my suggestion is:
* Take the current CandidateNoTest runtype code and repurpose it under a new runtype `LegacyImagesBuild`
* Implement CandidateNoTest to call the current Bazel Image Build + Push steps

**Comment by burmudar** [comment]
> > > NICE
> > > One question remains though, how do we rebuild those images outside of Bazel if we have to?
> > 
> > 
> > We could probably keep the current method that does around. Just have to rename it as it is currently `bazelBuildCandidateDockerImages` and maybe under different runType like `legacyDockerImages`
> 
> @jhchabran So this is exactly what the CandidateNoTest runtype does. So my suggestion is:
> 
> * Take the current CandidateNoTest runtype code and repurpose it under a new runtype `LegacyImagesBuild`
> * Implement CandidateNoTest to call the current Bazel Image Build + Push steps

I'll do this in a follow up PR

File: dev/oci_deps.bzl [hunk 3]
```diff
@@ -207,3 +207,40 @@ def oci_deps():
         digest = "sha256:dcc08b0228a5abd8a542efc7ee86ec58299591a737cafe19e92af6c1e1301c26",
         image = "index.docker.io/sourcegraph/wolfi-qdrant-base",
     )
+
+    # The following image digests are from tag 252535_2023-11-28_5.2-82b5f4f5d73f
```
Comment (jhchabran) [inline dev/oci_deps.bzl:211]: We need another comment here referrring to candidates-no-test

**Review by jhchabran — APPROVED** [review]
LGTM!

**Pull request merged**